trappy/cpu_power.py

```diff
@@ -120,6 +143,89 @@ def get_all_freqs(self, mapping_label):
 
         return pivot_with_labels(dfr, "freq", "cpus", mapping_label) / 1000
 
+    def plot_cdev_states(self, title="", width=None, height=None, xlim="default",
+                         ylim="range", legend_label="", cpus=None):
+        """Plot the cooling device state evolution
+
+        :param ax: Axis instance
+        :type ax: :mod:`matplotlib.Axis`
+
+        :param legend_label: Label for the legend
+        :type legend_label: str
+
+        :param title: The title of the plot
+        :type title: str
+
+        :param width: The width of the plot
+        :type width: int
+
+        :param height: The height of the plot
+        :type height: int
+
+        :param cpus: List of cpus to plot
+            All are plotted by default
+        :type tz_id: list of int
+        """
+        from matplotlib import pyplot as plt
+        from trappy.plot_utils import normalize_title, pre_plot_setup, post_plot_setup
```

Comment:
Um..  is this done elsewhere (importing inside a function)?

Reply:
Yeah I started off by copy/pasting the code for thermal:Thermal:plot_temperature which does this.

Reply:
OK with me then :man_shrugging: